hugo/content/docs/recipes/lexing/case-insensitive-languages.md

```diff
@@ -0,0 +1,54 @@
+---
+title: Case-insensitive languages
+weight: 100
+---
+
+Some programming languages such as SQL or Structured Text use case insensitivity to provide more flexibility when writing code. For example most SQL databases accept select statements starting with `select`, `SELECT` or even `SeLeCt`.
+
+In case you want to provide your users the same flexibility with your language, there are different levels of case-insensitivity in Langium:
+
+* You can make Langium's parser completely case insensitive using the language configuration
+* You can include case-insensitivity for specific terminal rules
+* You can make cross references case insensitive
+
+All of these options can be enabled independent of one another.
+
+## Case-insensitivity by configuration
+
+To make Langium case-insensitive, you have to set the `caseInsensitive` option to `true` in the `LangiumConfig` object which is located in the `langium-config.json` file at the root of your Langium project. You can set this up for every single language.
+
+```json5
+{
+    ...
+    "languages": [
+      {
+        "id": "hello-world",
+        "caseInsensitive": true, // <-- makes the specified language case insensitive
+        ...
+      },
+      ...
+    ],
+    ...
+}
+```
+
+## Case-insensitivity on demand
+
+If you want to include case-insensitivity only where you need it, you can use the `i` flag inside of your grammar's regular expressions
+
+```ts
```

Comment:
FYI, we support `langium` as a syntax highlighting configuration on our website
````suggestion
```langium
````